### PR TITLE
Fix openapi validator validateResponses config

### LINKED ISSUE #9733: `/api/v1/contracts/<address>/results/logs` throws `Internal Server Error`
### Description

Using version > `0.116.0`, the `/api/v1/contracts/<address>/results/logs` throws an error. 

On version <= `0.116.0`, the response is :point_down: which is expected.
```
{
  "_status":{
    "messages":[
      {
        "message":"No contract with the given evm address: {\"shard\":null,\"realm\":null,\"create2_evm_address\":{\"type\":\"Buffer\",\"data\":[85,85,85,85,85,85,85,85,85,85,85,85,85,85,85,85,85,85,85,85]}} has been found."
      }
    ]
  }
}
```

### Steps to reproduce

The current mainnet is running 0.116.1 and it's not broken - https://mainnet-public.mirrornode.hedera.com/api/v1/contracts/0x5555555555555555555555555555555555555555/results/logs

The current testnet is running 0.117.0 where it's broken - https://testnet.mirrornode.hedera.com/api/v1/contracts/0x5555555555555555555555555555555555555555/results/logs

### Additional context

_No response_

### Hedera network

_No response_

### Version

0.117.0

### Operating system

None

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

This PR fixes the `Internal Server Error` response on some REST API endpoints

- Fix openapi validator `validateResponses` config
- Change create2 evm address not found message
- Fix openapi spec

**Related issue(s)**:

Fixes #9733 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest/middleware/openapiHandler.js [hunk 1]
```diff
@@ -159,14 +159,13 @@ const serveSwaggerDocs = (app) => {
 };
 
 const openApiValidator = (app) => {
+  const validateResponses = isTestEnv() ? {allErrors: true} : false;
```
Comment (xin-hedera) [inline hedera-mirror-rest/middleware/openapiHandler.js:162]: we misunderstood what `validateResponses.allErrors` does:

```
allErrors: false | true
true - all rules should be checked and all failures reported
false - (default) stop checking rules after the first failure

File: hedera-mirror-rest/__tests__/specs/contracts/{id}/results/logs/not-found.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      }
+    },
+    {
+      "url": "/api/v1/contracts/0x5555555555555555555555555555555555555555/results/logs",
+      "responseStatus": 404,
+      "responseJson": {
+        "_status": {
+          "messages": [
+            {
+              "message": "No contract with the given evm address 0x5555555555555555555555555555555555555555 has been found."
```
Comment (xin-hedera) [inline hedera-mirror-rest/__tests__/specs/contracts/{id}/results/logs/not-found.json:22]: @natanasow The PR also changes the error message format. previously it'll be like

```
No contract with the given evm address: {\"shard\":null,\"realm\":null,\"create2_evm_address\":{\"type\":\"Buffer\",\"data\":[85,85,85,85,85,85,85,85,85,85,85,85,85,85,85,85,85,85,85,85]}} has been found.
```

**Review by edwin-greene — APPROVED** [review]
Looks good to me

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Pull request merged**